app/rpc/rpchandlers/estimate_network_hashes_per_second.go

```diff
@@ -11,7 +13,16 @@ func HandleEstimateNetworkHashesPerSecond(context *rpccontext.Context, _ *router
 	estimateNetworkHashesPerSecondRequest := request.(*appmessage.EstimateNetworkHashesPerSecondRequestMessage)
 
 	windowSize := int(estimateNetworkHashesPerSecondRequest.WindowSize)
-	networkHashesPerSecond, err := context.Domain.Consensus().EstimateNetworkHashesPerSecond(windowSize)
+	blockHash := model.VirtualBlockHash
+	if estimateNetworkHashesPerSecondRequest.BlockHash != "" {
+		var err error
+		blockHash, err = externalapi.NewDomainHashFromString(estimateNetworkHashesPerSecondRequest.BlockHash)
+		if err != nil {
+			return nil, err
```

Comment:
This would cause kaspad to crash. Since a failure in NewDomainHashFromString can only be a parsing error, please respond with an appropriate EstimateNetworkHashesPerSecondResponseMessage instead.